### PR TITLE
(chore) Improve 3rd party language quick start guide

### PR DESCRIPTION
**Pull request by jf990**
Proposing some changes to https://github.com/highlightjs/highlight.js/blob/master/extra/3RD_PARTY_QUICK_START.md

I worked out the process while working on https://github.com/highlightjs/highlightjs-curl, so that repo and this doc are in alignment, as well as the proposed template repo https://github.com/jf990/highlightjs-language-template

based on a review here there are a few final changes to complete, but wanted to get some feedback before finishing.

### Changes

Update https://github.com/highlightjs/highlight.js/blob/master/extra/3RD_PARTY_QUICK_START.md to improve the doc.

### Checklist

- [ ] ~~Added markup tests, or they don't apply here because...~~ (doc update)
- [ ] Updated the changelog at `CHANGES.md` TBD

### CODE REVIEW

File: extra/3RD_PARTY_QUICK_START.md
Comment (joshgoebel) [inline extra/3RD_PARTY_QUICK_START.md:34]: Is that what people are doing?  It's much easier to just have/move your own repo's `git checkout` directly inside the extra folder... no need to copy or link anything... I wouldn't recommend copying.

If symlinking the folder into extra works we might recommend that (I haven't tested).

File: extra/3RD_PARTY_QUICK_START.md
Comment (joshgoebel) [inline extra/3RD_PARTY_QUICK_START.md:30]:
```suggestion
For example, if you your grammar is named `pascal` you would create your repo directory structure as follows:

- Put your grammar file in `src/languages/pascal.js`.
- Add detect tests in `test/detect/pascal/`. 
- Add markup tests in `test/markup/pascal/`.
- Add a `package.json` file.
- Add a LICENSE so people know how your work is licensed.
```

File: extra/3RD_PARTY_QUICK_START.md
Comment (joshgoebel) [inline extra/3RD_PARTY_QUICK_START.md:18]: This should be reworded to encourage people to start with their own repo - we can migrate it to our organization later if the user requests.  What happens when we let them decide first is we end up having a few people who never finish their grammar and then we have dead repos in our organization for a long time - and it's hard to decide when they should be removed.  Just awkward.

Would be much nicer to have someone send a link to their repo all shiney and ready to go and then if everything looks good it can be transferred or forked, etc... I think that's the sequencing we'd like to head towards.

CC @allejo

File: extra/3RD_PARTY_QUICK_START.md
Comment (joshgoebel) [inline extra/3RD_PARTY_QUICK_START.md:74]: Did you cross this out because your feel the discussion is over, or?

File: extra/3RD_PARTY_QUICK_START.md
Comment (jf990) [reply extra/3RD_PARTY_QUICK_START.md:34]: I don't know about the git checkout process inside an existing repo, I need to learn how to do that. If that works it solves a major issue I have with this process and how I approached the set up for the template project.

File: extra/3RD_PARTY_QUICK_START.md
Comment (jf990) [reply extra/3RD_PARTY_QUICK_START.md:30]: I agree with your suggestion, but as you also mentioned i think we should use a non-language file here because I wanted it to be the exact file used in the template project, and then very obvious all the files you need to rename. Should we go back to `xyz`?

File: extra/3RD_PARTY_QUICK_START.md
Comment (jf990) [reply extra/3RD_PARTY_QUICK_START.md:74]: related to your comment above, my initial approach was start with a highlightjs-xyz repo firs. but I agree, the better process is start with your own repo first, and transfer to highlightjs after approval. I'll update the docs to promote this process instead of what I had.

File: extra/3RD_PARTY_QUICK_START.md
Comment (joshgoebel) [reply extra/3RD_PARTY_QUICK_START.md:34]: Well git checkout or symlink (for any Unix people)... I never tried symlinking but can't imagine why it wouldn't work.

File: extra/3RD_PARTY_QUICK_START.md
Comment (joshgoebel) [reply extra/3RD_PARTY_QUICK_START.md:30]: `xyz` is ok I think, though a placeholder is easy enough to search/replace...

File: extra/3RD_PARTY_QUICK_START.md
Comment (allejo) [reply extra/3RD_PARTY_QUICK_START.md:18]: I agree with Josh. Having a complete and functional grammar would at least allow hljs maintainers to approve PRs should it be abandoned. But if it's incomplete, then it'd be up to the maintainers to complete it.

Comment (jf990) [reply extra/3RD_PARTY_QUICK_START.md:21]: including a reference to the template should make this easier

**Comment by jf990** [comment]
updated and tested inside `extra` as a cloned repo and updated assuming we are going to move forward with the template repo. @joshgoebel @allejo suggestions applied.

File: extra/3RD_PARTY_QUICK_START.md
Comment (joshgoebel) [inline extra/3RD_PARTY_QUICK_START.md:66]: Why?  There should be nothing to copy.  If their repo is under extra then the file is already in the dist folder exactly where it belongs, it only needs to be committed.

**Comment by joshgoebel** [comment]
Under testing we should mention `ONLY_EXTRA` env which will more tightly scope the test to only running against languages inside `extra` rather than all the core languages.  This only works for markup tests, but those are the most common that need to be run over and over.

```
ONLY_EXTRA=true npm run test-markup
```

**Comment by jf990** [comment]
@joshgoebel suggestions are up. thanks!

**Review by joshgoebel — APPROVED** [review]

**Pull request merged**
